Added SpecTableLayout

webwarrior-ws opened the pull request:
That's it.

If you have any remarks about following Spec conventions, or if something's missing, let me know.

Also I have to note code duplication in `#subwidget:fractions:offsets:` and `#subwidget:layoutFrame:` methods of `SpecLayoutAdd`, and now also in `#subwidget:` method in `SpecTableLayoutAdd`. I don't know what you do in Smalltalk in such situations (create a trait?), but common code needs to be extracted.

webwarrior-ws committed:
Added SpecTableLayout

webwarrior-ws committed:
Recovered TableContainerModel class, lost during merge

BenjaminVanRyseghem commented:
Thanks for the PR.
I am actually in the process of moving out (putting everything in boxes, getting shout on by my wife because it's not fast enough, etc ;) ) so I will review your PR asap, but it may take some time.
Sorry in advance, and thanks again

BenjaminVanRyseghem commented on Spec-Core.package/SpecInterpreter.class/instance/computeSpecFrom.selector..st:
```diff
@@ -5,7 +5,7 @@ computeSpecFrom: aSymbol selector: aSelector
 		ifTrue: [ | result |
 			result := self convertSymbolOfClassToInstance: aSymbol.
-			aSymbol == #ContainerModel
+			(result isKindOf: ContainerModel)
```
I think it would be better to implement a message like #isContainer on ContainerModel and ComposableModel.
I tend to avoid as much as possible usages of #isKindOf

BenjaminVanRyseghem commented on Spec-Layout.package/SpecTableLayout.class/instance/width..st:
```diff
@@ -0,0 +1,4 @@
+commands
+width: aNumber
+	"Doesn't work when opened in window, only when part of another layout"
```
any clue why?

webwarrior-ws replied:
Window enforces its size to its contents, and not the other way round.

BenjaminVanRyseghem commented on Spec-Layout.package/SpecTableLayoutAddWithSpec.class/instance/subwidget.spec..st:
```diff
@@ -0,0 +1,7 @@
+	self subwidget: (sub isArray 
+						ifTrue: [ #(model), sub , {#retrieveSpec:. aSpecSelector} ]
+						ifFalse: [ {#model. sub . #retrieveSpec: . aSpecSelector } ]).
+	
```
Strange indentation here

webwarrior-ws replied:
Seems to be different tab sizes. I'll fix the formatting.

BenjaminVanRyseghem commented:
Nice pull request :)

Do we wait for a fix before merging it? :)

webwarrior-ws commented:
Fix will be ready soon (about tomorrow), so why not to wait?

BenjaminVanRyseghem commented:
Perfect :)
I will be waiting for your fix then :)

Thanks again :)

webwarrior-ws committed:
Minor fixes considering remarks to pull request #15

Changed container detection in SpecInterpreter (now uses
#isSpecContainer).
Fixed indentation in SpecTableLayoutAddWithSpec>>#subwidget:spec

BenjaminVanRyseghem commented:
Sorry I hadn't seen you had pushed a fix :S
